### ISSUE TITLE
bn_param layer appears to be missing

### ISSUE THREAD

**Issue by nk-dev0** [jimchenhub/new-caffe-segnet#2]
[libprotobuf ERROR google/protobuf/text_format.cc:274] Error parsing text-format caffe.NetParameter: 51:12: Message type "caffe.LayerParameter" has no field named "bn_param".

It looks like bn_layer.cpp and *.cu are missing from your src/layers folder

Do I need to change the bn layer to batch_normalization or whatever the newer version of caffe uses in the prototxt?

**Comment by nk-dev0** [comment]
Or is it possible to create a new segnet_train.prototxt (from https://github.com/alexgkendall/SegNet-Tutorial/blob/master/Models/segnet_train.prototxt) using the new caffe version since I've changed all "bn_param" to "batch_norm_param" but now I'm getting the error:

    Error parsing text-format caffe.NetParameter: 52:22: Message type "caffe.BatchNormParameter" has no field named "scale_filler".

**Comment by jimchenhub** [comment]
There is no BN layer in source code. I didn't use this layer but achieve better result.

Caffe use Batch normalization with two layers: `BatchNorm ` and `Scale`
reference: http://stackoverflow.com/documentation/caffe/6575/batch-normalization#t=201612100645062427067
